agent/app/agent_compatibility_linux.go

```diff
@@ -0,0 +1,48 @@
+// +build linux
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package app
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/aws/amazon-ecs-agent/agent/engine"
+	"github.com/aws/amazon-ecs-agent/agent/utils"
+	"github.com/cihub/seelog"
+	"github.com/pkg/errors"
+)
+
+func (agent *ecsAgent) checkCompatibility(engine engine.TaskEngine) error {
```

Comment:
Can this method also verify that the agent has access to the cgroup mounts?

Reply:
I'm not sure that fits within the scope of responsibility for this function. Here we are checking the state file for incompatibilities. 

Don't we handle that error in the agent startup process (with cgroups enabled)?

Reply:
The agent does not do this yet. We only have a `TODO`. Have a look at `L#196`

Reply:
aha! I'll need to think about that.

We want that to happen in more cases than only when this function gets invoked.

Reply:
I think a better place for the `mountpoint` check would be the `config` package as we can stick it in the `overrides`

Reply:
I agree that we should do that check outside of this function, possibly somewhere in this block: https://github.com/aws/amazon-ecs-agent/pull/1007/files#diff-2149975f54c050dd31ac9a37f1ef63bcR198

Reply:
Agreed. I'll address that separately.